### ISSUE TITLE
integrate and compare with bioregistry

### ISSUE THREAD

**Issue by cmungall** [prefixcommons/biocontext#38]
This does most of what we set out to do with this repo and more: https://github.com/bioregistry/bioregistry

There are still some things this repo does that bioregistry doesn't do, but these could probably either be moved there OR this repo could be hollowed out and made more of a simple overlay

 - generating JSON-LD contexts from one or more upstream sources
 - inclusion of standard non-bio semweb prefixes (rdf, rdfs, owl, xsd, ...)
 - allow people to remix their own prefix mappings, choosing priority lists in the event of clashes

Note that there is a ticket for integrating prefixcommons:
https://github.com/bioregistry/bioregistry/issues/9

But this repo (biocontexts) is distinct from the repo that was integrated... we seem to have caused some confusion here cc @nmatentz @cthoyt

**Comment by cthoyt** [comment]
@cmungall I've included a few semantic web prefixes already. Would you be willing to try adding the rest yourself and report back on how good the documentation is at helping you get there?

I'm open to implementing the new features as well

**Comment by cmungall** [comment]
Should I go ahead and add directly to your bioregistry.json?

One thing that wasn't clear to me how to deal with forks or incorporating changes from upstream registries that may conflict with things curated into bioregistry.json

**Comment by cthoyt** [comment]
Yes, feel free to make a PR directly in the https://github.com/bioregistry/bioregistry/edit/main/src/bioregistry/data/bioregistry.json file. Put it wherever you want, there's a linter that will reorganize it

**Comment by cthoyt** [comment]
I updated the way the bioregistry handles preferred cases and added code to automatically generate the JSON-LD context file. It's available here and will update nightly https://github.com/bioregistry/bioregistry/blob/main/docs/_data/contexts/obo_context.jsonld.

**Comment by mellybelly** [comment]
@jmcmurry can we look at some of the nested prefix handling?

**Comment by jmcmurry** [comment]
@mellybelly I'm missing some context here. Maybe you mean indigenous vs expat references? eg. MGI:00001 vs monarch:MGI:00001 I'm not convinced that nesting the prefixes is necessarily the best way to display this as it doesn't scale gracefully. That said there could be a good use case for documenting incoming IDs like so: https://github.com/monarch-initiative/dipper/blob/master/dipper/incoming_curie_map.yaml and that could ultimately drive resolution behavior.

**Comment by mellybelly** [comment]
comment was not about nested identifiers but rather resolution according to nested structured metadata... trying to recall where we documented these examples. maybe @jkunze recalls, maybe in NT2

**Comment by jmcmurry** [comment]
That doesn't ring a bell but end users can already choose to resolve based on default (up time) or by hard coding a specific provider. https://docs.identifiers.org/articles/docs/resolving_mechanisms.html
